spec/controllers/tag_set_nominations_controller_spec.rb

```diff
@@ -0,0 +1,1547 @@
+                      expect(assigns(:nominations_count)[:character]).to eq(1)
+                      expect(assigns(:nominations)[:character].count).to eq(1)
+                      expect(assigns(:nominations_count)[:relationship]).to eq(31)
+                      expect(assigns(:nominations)[:relationship].count).to eq(30)
+                    end
+
+                    it 'returns a flash notice about too many nominations' do
+                      expect(flash[:notice]).to eq("There are too many nominations to show at once, so here's a " +
+                                                       "randomized selection! Additional nominations will appear " +
+                                                       "after you approve or reject some.")
+                    end
+                  end
+                end
+
+                context 'character_nomination_limit is 0' do
+                  it 'does not return character nominations' do
+                    add_unreviewed_character_nominations(1)
+                    owned_tag_set.update_column(:character_nomination_limit, 0)
+                    get :index, tag_set_id: owned_tag_set.id
+                    expect(assigns(:nominations)[:character]).to be_nil
+                  end
+                end
+
+                context 'relationship_nomination_limit is 0' do
+                  it 'does not return relationship nominations' do
+                    add_unreviewed_relationship_nominations(1)
+                    owned_tag_set.update_column(:relationship_nomination_limit, 0)
+                    get :index, tag_set_id: owned_tag_set.id
+                    expect(assigns(:nominations)[:relationship]).to be_nil
+                  end
+                end
+
+                def add_unreviewed_character_nominations(num)
+                  num.times do |i|
+                    CharacterNomination.create(tag_set_nomination: tag_set_nomination, fandom_nomination: fandom_nom,
+                                               tagname: "New Character #{i}")
+                  end
+                end
+
+                def add_unreviewed_relationship_nominations(num)
+                  num.times do |i|
+                    RelationshipNomination.create(tag_set_nomination: tag_set_nomination, fandom_nomination: fandom_nom,
+                                                  tagname: "New Relationship #{i}")
+                  end
+                end
+              end
+            end
+          end
+
+          context 'logged in user is not moderator' do
+            let(:user) { random_user }
+
+            it 'redirects and returns an error message' do
+              get :index, tag_set_id: owned_tag_set.id
+              it_redirects_to_with_error(tag_sets_path, "You can't see those nominations, sorry.")
+            end
+          end
+        end
+
+        # TODO: how can OwnedTagSet.find not raise an error but still return falsey?
```

Comment:
I think this is probably hitting the error message in load_tag_set rather than in load_nomination, so the error should be "What tag set did you want to nominate for?" ?